zerver/tests/test_slack_importer.py

```diff
@@ -416,14 +453,42 @@ def test_convert_slack_workspace_messages(self, mock_get_total_messages_and_user
         zerver_message2 = [{'id': 5}]
 
         realm = {'zerver_subscription': []}  # type: Dict[str, Any]
+        user_list = []  # type: List[Dict[str, Any]]
 
         zerver_usermessage1 = [{'id': 3}, {'id': 5}]
         zerver_usermessage2 = [{'id': 6}, {'id': 9}]
 
         mock_message.side_effect = [[zerver_message1, zerver_usermessage1],
                                     [zerver_message2, zerver_usermessage2]]
-        message_json = convert_slack_workspace_messages('./random_path', 2, {},
+        message_json = convert_slack_workspace_messages('./random_path', user_list, 2, {},
                                                         {}, added_channels,
                                                         realm)
         self.assertEqual(message_json['zerver_message'], zerver_message1 + zerver_message2)
         self.assertEqual(message_json['zerver_usermessage'], zerver_usermessage1 + zerver_usermessage2)
+
+    @slow('Imports converted slack data into existing database.')
+    @mock.patch("zerver.lib.slack_data_to_zulip_data.get_user_data")
+    def test_slack_import_to_existing_database(self, mock_get_user_data: mock.Mock) -> None:
```

Comment:
This test takes approx 13 seconds. We have a separate db for testing the slack importer(https://github.com/zulip/zulip/blob/master/zproject/slack_importer_test_settings.py), However this would defeat the purpose of testing the import into a db with already existing data.

Reply:
That's really slow (like, 1/4 of our total test runtime slow).  Do you understand why it's so slow?  It seems like this is something we'll want to optimize.